### PR TITLE
canary command

### LINKED ISSUE #345: --canary flag
**Is your feature request related to a problem? Please describe.**

I would like to be able to release a canary version through auto on each PR.

**Describe the solution you'd like**

The ability to run:

```sh
auto shipit --canary
```

Which publishes a version like `auto@1.2.3-canary.PR_NUMBER.BUILD_NUMBER`

**Describe alternatives you've considered**

Manually running lerna or npm commands, ew. 😉

### PR DESCRIPTION
**Pull request by hipstersmoothie**
# What Changed

Add the --canary flag to `shipit`

# Why

closes #345 

Todo:

- [x] Add tests
- [x] Add docs
- [ ] Add yourself to contributors (run `yarn contributors:add`)

### CODE REVIEW

**Comment by adierkens** [comment]
Published PR with canary version: `4.10.1-canary.349.4151`

**Comment by adierkens** [comment]
Published PR with canary version: `4.10.1-canary.349.4157`

**Comment by adierkens** [comment]
Published PR with canary version: `4.10.1-canary.349.4161`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.349.4182`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.349.4187`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.349.4192`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.undefined.undefined`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.349.4217`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.349.4222`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.349.4227`

**Comment by adierkens** [comment]
Published PR with canary version: `4.12.0-canary.349.4232`

File: src/plugins/npm/index.ts [hunk 1]
```diff
@@ -248,7 +249,6 @@ export default class NPMPlugin implements IPlugin {
         'NPM - Monorepo',
         async (commits, renderLine) => {
           if (!isMonorepo()) {
-            console.log('bailed');
```
Comment (zephraph) [inline src/plugins/npm/index.ts]: 👍 I'd been seeing this in the logs and wondered what it was, ha.

**Comment by adierkens** [comment]
:rocket: PR was released in v4.12.0 :rocket:

**Pull request merged**